src/traits.rs

```diff
@@ -0,0 +1,29 @@
+use {
+    crate::{TextRange, TextSize},
+    std::convert::TryInto,
+};
+
+/// Text-like structures that have a text size.
+pub trait TextSized {
```

Comment:
We could seal/voldemort this trait if desired, but I think it's probably worth allowing/encouraging downstream users to impl this trait for their own type (reference)s that encapsulate some text (e.g. `rowan::green::{Node, Token}`).

Like `FromStr`, however, the standard way to use the trait isn't through the trait, but rather through the accessor that's implemented in terms of it: `TextSize::of` (`str::parse`).

Reply:
Yeah, that's my thinking as well: this trait exist and is public for extending the library, but all the API is pinned to `TextSize`, `TextRange`, so that you only need to import two concrete types.

I am also not sue about the name of the trait. Name collision is bad... Given that this is for extensions, perhaps we can go with uglier `LenTextSize / len_text_size`, to match `len_utf()`?